src/Deserialization/Transaction.js

```diff
@@ -84,13 +91,40 @@ exports._convertCert = certConvHelper => cert => {
     // StakeRegistration,
     var r = cert.as_stake_registration();
     if (r) return certConvHelper.stakeRegistration(r.stake_credential());
```

Comment:
I think it would be better to use a switch block here in conjunction with [CSL.CertificateKind](https://input-output-hk.github.io/cardano-js-sdk/enums/_cardano_sdk_core.CSL.CertificateKind.html) and [Certificate.Kind](https://input-output-hk.github.io/cardano-js-sdk/classes/_cardano_sdk_core.CSL.Certificate.html#kind)

Reply:
We don't use cardano SDK. These are `cardano-serialization-lib` bindings. Here `kind()` outputs a number (no named constants are provided). I don't like using magic numbers in code, plus the overhead is constant and negligible.

Reply:
Those docs (from what I've observed) are 1 to 1, so the enum should be available. It's not really for performance, I think it would make it significantly more readable. We can entirely forgo the as_ and use the value directly

Reply:
(Just did a quick sanity check and `lib.CertificateKind.GenesisKeyDelegation` for eg is indeed available)

Reply:
Ah, I don't think as_* are simple kind checks and type coercion, ie you still need to call it first